internal/builders/bazel/rebuilder.sh

```diff
@@ -0,0 +1,490 @@
+# The binaries folder contains different directories for the its artifacts and
+# the artifacts runfiles. Obtain the rebuilt binaries and copy them to the
+# path at root before cleaning up and deleting the repo.
+if [[ "$artifact_name" == *"_deploy.jar"* ]]
+then
+      # Uses _deploy.jar as a field seperator and grabs the field before it.
+      # Directory of Java artifacts is same as run script name.
+      run_script_name=$(echo "$artifact_name" | awk -F'_deploy.jar' '{print $1}')
+      cd $binaries_dir/
+      rebuilt_checksum=$(sha256sum ./"$run_script_name"/"$artifact_name" | awk '{ print $1 }')
+
+      # Copy the entire directory, including the run script.
+      cp -R ./"$run_script_name" ./../../"$rebuilt_artifacts_dir"/
+else
+    if [[ "${NEEDS_RUNFILES}" == "true" ]]
+    then
+        # For non-java targets with runfiles.
+      cd $binaries_dir/
+      rebuilt_checksum=$(sha256sum ./"$artifact_name"/"$artifact_name" | awk '{ print $1 }')
+
+      # Copy entire directory, including the runfiles.
+      cp -R ./"$artifact_name" ./../../"$rebuilt_artifacts_dir"/
+    else
+    # For files withouts runfiles.
+    cd $binaries_dir
+    rebuilt_checksum=$(sha256sum "$artifact_name" | awk '{ print $1 }')
+
+    cp "$artifact_name" ./../../$rebuilt_artifacts_dir/
+    fi
+fi
+
+################################################
+#                                              #
+#       Check Build for Reproducibility        #
+#                                              #
+################################################
+
+if [[ "$orig_checksum" == "$rebuilt_checksum" ]]
+then
+    printf "${GREEN}Checksum is the ${BOLD}${UNDERLINE}same${RESET}${GREEN} for the original and rebuilt artifact!${RESET}\n"
+    printf "${GREEN}✅ This build is ${BOLD}${UNDERLINE}reproducible!${RESET} ✅ \n"
+    echo ""
+    printf "${GREEN}%s${RESET} = Original Checksum${RESET}\n" "$orig_checksum"
+    printf "${GREEN}%s${RESET} = Rebuilt Checksum${RESET}\n" "$rebuilt_checksum"
+    echo ""
+else
+    printf "${BOLD_RED_BG}Checksum is ${BOLD}${UNDERLINE}NOT${RESET}${BOLD_RED_BG} the same for the original and rebuilt artifact!${RESET}\n"
+    printf "${BOLD_RED_BG}        ⚠️  This build was ${BOLD}${UNDERLINE}NOT${RESET}${BOLD_RED_BG} able to be reproduced! ⚠️         ${RESET}\n"
+    echo ""
+    printf "${RED}%s${RESET} = Original Checksum\n" "$orig_checksum"
+    printf "${RED}%s${RESET} = Rebuilt Checksum\n" "$rebuilt_checksum"
+    echo ""
+fi
+
+
+if [[ cleanup -eq 1 ]]
+then
+
+  # If there are runfiles or if the artifacts are Java, then each artifact
+  # has its own directory, so you need to exit out of it first.
+  if [[ "${NEEDS_RUNFILES}" == "true" || "${INCLUDES_JAVA}" == "true" ]]
+  then
+    cd ..
+  fi
+
+  # Current position is bazel_builder_dir/$repo_name/$binaries_dir,
+  # and to clean up need to be in /bazel.
+  cd ../..
+
+  # Now cleanup of verifier and cloned $repo_name.
+  cleanup
```

Comment:
@ianlewis how do you feel about this huge script? I'm worried it will be hard to maintain. Shall we use another language?

Reply:
I'm not against it. 500 lines isn't *that* long. But I do think it could be simplified depending on how it's used. Do we need all the command line parsing stuff for example?

@enteraga6 I didn't see where this script was executed? Is it done via the bazel workflow somehow?

Reply:
Don't know why I didn't reply to this thread but replied to review instead..

My response down there was: 
@ianlewis This script is not executed in a workflow. It is executed on the local machine of the user.

I will switch script to have utils and source it at beginning.